### PR TITLE
Fix kafka storage does not work with parquet format messages

### PR DESCRIPTION
**Pull request by godliness**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Kafka storage may support parquet format messages

Detailed description / Documentation draft:

Before:

`prepareReader` create `file_reader` before kafka messages received and available

After:

Put `prepareReader` into `generate`, when `generate` called  kafka messages should be available

### CODE REVIEW

**Review by filimonov — APPROVED** [review]
Lgtm. But check tests

**Comment by godliness** [comment]
> Lgtm. But check tests

Thanks @filimonov,

Other failed tests are not related to this PR.

`test_kafka_engine_put_errors_to_stream_with_random_malformed_json`  test failed is only for format `JSONEachRow`, so i think it's also not relate to this PR.

PTAL

**Comment by alexey-milovidov** [comment]
Need help from @filimonov to fix flaky test.

**Comment by kitaisreal** [comment]
@filimonov can you take a look at broken asan test ?

**Comment by godliness** [comment]
@kitaisreal @filimonov 

Hello friends, what should i do next? ^ _ ^

**Comment by filimonov** [comment]
Sorry go delay @godliness 

The failure does not look related to your changes (rather test is not stable
https://github.com/ClickHouse/ClickHouse/pull/21850 ), but i still need to try to figure out how to fix it before merge. Had no time yet, but will try to look on that today.

File: tests/integration/test_storage_kafka/test.py [hunk 1]
```diff
@@ -574,39 +574,13 @@ def test_kafka_formats(kafka_cluster):
...
+            ],
```
Comment (filimonov) [inline tests/integration/test_storage_kafka/test.py:582]: I guess with empty messages it will still break, but it's acceptable.

**Review by filimonov — APPROVED** [review]

File: tests/integration/test_storage_kafka/test.py (hunk 1 above)
Comment (godliness) [reply tests/integration/test_storage_kafka/test.py:582]: yes, agree it, empty message size is zero also.

**Comment by filimonov** [comment]
@Mergifyio update

**Comment by filimonov** [comment]
@kitaisreal the rest of failures are not Kafka related.

**Pull request merged**